src/wp-includes/embed.php

```diff
@@ -411,10 +411,17 @@ function get_post_embed_url( $post = null ) {
 		return false;
 	}
 
-	$embed_url     = trailingslashit( get_permalink( $post ) ) . user_trailingslashit( 'embed' );
-	$path_conflict = get_page_by_path( str_replace( home_url(), '', $embed_url ), OBJECT, get_post_types( array( 'public' => true ) ) );
+	$is_published = 'publish' === $post->post_status;
 
-	if ( ! get_option( 'permalink_structure' ) || $path_conflict ) {
+	if ( $is_published ) {
+		$embed_url = trailingslashit( get_permalink( $post ) ) . user_trailingslashit( 'embed' );
+
+		$path_conflict = get_page_by_path( str_replace( home_url(), '', $embed_url ), OBJECT, get_post_types( array( 'public' => true ) ) );
```

Comment:
Since this is quite a bit of information, I'd recommend adding some linebreaks here. 

Indents may be wrong in my suggestion, since I just add this via GitHub.

```suggestion
		$path_conflict = get_page_by_path(
			str_replace( home_url(), '', $embed_url ), 
			OBJECT, 
			get_post_types( array( 'public' => true ) ) 
		);
```

For me this makes the difference between "wait what belongs where" and "ahh ok, understood"

Reply:
Fixed